### PR TITLE
Link support for notebook output. re microsoft/vscode-jupyter#12285

### PR DESCRIPTION
**Pull request by rebornix**
Adding back link support.

microsoft/vscode-jupyter#12285

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts [hunk 1]
```diff
@@ -894,6 +877,76 @@ var requirejs = (function() {
 		}));
 	}
 
+	private _handleResourceOpening(href: string) {
+		let linkToOpen: URI | undefined = undefined;
+		if (href.startsWith('/')) {
+			linkToOpen = URI.parse(href);
```
Comment (mjbvz) [inline src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts:883]: Should this be URI.file instead?

File: src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts [hunk 2]
```diff
@@ -894,6 +877,76 @@ var requirejs = (function() {
...
+
+		if (linkToOpen) {
+			this._openUri(linkToOpen);
+		}
+	}
+
+	private _openUri(uri: URI) {
+		let lineNumber: number | undefined = undefined;
+		let column: number | undefined = undefined;
+		const lineCol = LINE_COLUMN_REGEX.exec(uri.path);
```
Comment (mjbvz) [inline src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts:910]: Is the line number really in the path and not in the `fragment` or `query` in this case?

If it is, is this a path syntax that only jupyter supports?

File: src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts [hunk 3]
```diff
@@ -894,6 +877,76 @@ var requirejs = (function() {
...
+				path: uri.path.slice(0, lineCol.index),
+				fragment: `L${lineCol[0].slice(1)}`
+			});
+			lineNumber = parseInt(lineCol[1], 10);
+			column = parseInt(lineCol[2], 10);
+		}
+
+		//#region error renderer migration, remove once done
+		const lineMatch = LineQueryRegex.exec(uri.query);
+		if (lineMatch) {
```
Comment (mjbvz) [inline src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts:922]: Check that `lineNumber` is a valid number here using `!isNaN`

**Review by mjbvz — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts (hunk 1 above)
Comment (rebornix) [reply src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts:883]: They are identical in this case.

**Review by mjbvz — APPROVED** [review]

**Pull request merged**